test/cpp/naming/resolver_component_test.cc

```diff
@@ -329,24 +340,22 @@ void test_resolves(grpc_exec_ctx *exec_ctx, args_struct *args) {
                             &on_resolver_result_changed);
 
   grpc_exec_ctx_flush(exec_ctx);
-  poll_pollset_until_request_done(args);
+  PollPollsetUntilRequestDone(args);
   GRPC_RESOLVER_UNREF(exec_ctx, resolver, NULL);
 }
 
-void resolver_component_test_resolution(
-    std::string target_name, std::vector<GrpcLBAddress> expected_addrs,
-    const char *expected_service_config, const char *expected_lb_policy) {
+TEST(ResolverTest, ResolvesRelevantRecords) {
   grpc_init();
-  gpr_log(GPR_INFO, "expected addresses size: %d", (int)expected_addrs.size());
   grpc_exec_ctx exec_ctx = GRPC_EXEC_CTX_INIT;
   args_struct args;
   args_init(&exec_ctx, &args);
-  args.target_name = target_name;
-  args.expected_addrs = expected_addrs;
-  args.expected_service_config_string = expected_service_config;
-  args.expected_lb_policy = expected_lb_policy;
+  args.expected_addrs = ParseExpectedAddrs(FLAGS_expected_addrs);
+  args.expected_service_config_string =
+      FLAGS_expected_chosen_service_config.c_str();
```

Comment:
If you save this as a C++ string instead of a C string, then you can change `check_service_config_result_locked()` to just use `EXPECT_EQ()`.

(Actually, even if you keep it as a C string, you can use `EXPECT_STREQ()`.)